pkg/apis/pipeline/v1beta1/pipeline_conversion.go

```diff
@@ -259,13 +259,37 @@ func (m *Matrix) convertTo(ctx context.Context, sink *v1.Matrix) {
 		param.convertTo(ctx, &new)
 		sink.Params = append(sink.Params, new)
 	}
+	if m.hasInclude() {
+		for _, include := range m.Include {
+			sink.Include = append(sink.Include, v1.MatrixInclude{Name: include.Name})
+			for _, param := range include.Params {
+				newIncludeParam := v1.Param{}
+				param.convertTo(ctx, &newIncludeParam)
+				sink.Include[len(sink.Include)-1].Params = append(sink.Include[len(sink.Include)-1].Params, newIncludeParam)
+			}
+		}
+	}
 }
 
 func (m *Matrix) convertFrom(ctx context.Context, source v1.Matrix) {
 	for _, param := range source.Params {
 		new := Param{}
 		new.convertFrom(ctx, param)
 		m.Params = append(m.Params, new)
+		if m.hasInclude() {
+			for _, include := range m.Include {
+				include.Params = append(include.Params, new)
```

Comment:
This feels like not changing the actuall Matrix object. so we could remove the iteration here